### PR TITLE
Support rabbitmq plugin

### PR DESCRIPTION
**Pull request by WithLin**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues

___
### Bug fix
- Bug description.

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

### CODE REVIEW

File: apm-protocol/apm-network/src/main/java/org/apache/skywalking/apm/network/trace/component/ComponentsDefine.java [hunk 1]
```diff
@@ -100,6 +100,10 @@
 
     public static final OfficialComponent UNDERTOW =  new OfficialComponent(49, "Undertow");
 
+    public static final OfficialComponent RABBITMQ_PRODUCER = new OfficialComponent(52,"rabbitmq-producer");
```
Comment (wu-sheng) [inline apm-protocol/apm-network/src/main/java/org/apache/skywalking/apm/network/trace/component/ComponentsDefine.java]: Could you put a comment here, why 50 and 51 should keep in empty.

File: apm-sniffer/apm-sdk-plugin/pom.xml [hunk 2]
```diff
@@ -60,6 +60,7 @@
         <module>activemq-5.x-plugin</module>
         <module>elasticsearch-5.x-plugin</module>
         <module>undertow-plugins</module>
+		<module>rabbitmq-5.x-plugin</module>
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/pom.xml]: Format is not right.

File: apm-sniffer/apm-sdk-plugin/rabbitmq-5.x-plugin/src/test/java/org/apache/skywalking/apm/plugin/rabbitmq/RabbitMQConsumerInterceptorTest.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    private  RabbitMQConsumerInterceptor rabbitMQConsumerInterceptor;
+
+    private Object[] arguments;
+
+    @Before
+    public void setUp() throws Exception {
+        rabbitMQConsumerInterceptor = new RabbitMQConsumerInterceptor();
+        Envelope envelope = new Envelope(1111,false,"","rabbitmq-test");
+        Map<String, Object> headers = new HashMap<String, Object>();
+        headers.put("sw6","");
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/rabbitmq-5.x-plugin/src/test/java/org/apache/skywalking/apm/plugin/rabbitmq/RabbitMQConsumerInterceptorTest.java]: Why empty header?

**Comment by wu-sheng** [comment]
@WithLin we got 8 passes, and 4 fails. Could you recheck? Look like the 4 of 5 latest versions fail.

**Comment by WithLin** [comment]
>we got 8 passes, and 4 fails. Could you recheck? Look like the 4 of 5 latest versions fail.

In view of this problem,i think that  the latest  version will change a lot.It may be supported in multiple versions. wait a minute. I'll check it again.

**Comment by wu-sheng** [comment]
Sure. If you are facing conflict, remember, you could use witness class to identify versions.

**Comment by WithLin** [comment]
I've adjusted my mistakes, so you can run the test case again.

**Comment by wu-sheng** [comment]
Thank @ascrutae to address the bug in this PR.

**Comment by wu-sheng** [comment]
@WithLin Please make the compatible better. At least, 5.4.x and 5.5.x should be supported. Thanks.

**Comment by WithLin** [comment]
I adjusted the test case again, and then ran all versions of rabbitmq in my local environment, all compatible. Please run the test case again. @wu-sheng

**Pull request merged**